keras_preprocessing/image/utils.py

```diff
@@ -130,7 +130,7 @@ def load_img(path, grayscale=False, color_mode='rgb', target_size=None,
 def list_pictures(directory, ext='jpg|jpeg|bmp|png|ppm'):
     return [os.path.join(root, f)
             for root, _, files in os.walk(directory) for f in files
-            if re.match(r'([\w]+\.(?:' + ext + '))', f.lower())]
+            if re.match(r'([\w\d-]+\.(?:' + ext + '))', f.lower())]
```

Comment:
The problem is not in the regex but in the use of `match`, as it only matches at the beginning. Also the main use of the function is the check of the extensions. There are easier and clearer ways to do this, like `.endswith`.